src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -263,6 +267,10 @@ SymbolKind.NetModule or
                 // At this point, we now absolutely should have our tree in the compilation
                 Contract.ThrowIfFalse(compilationPair.CompilationWithoutGeneratedDocuments.ContainsSyntaxTree(tree));
 
+                // Mark whatever generator state we have as not only final, but frozen as well.  We'll want to keep
+                // whatever we have here through whatever future transformations occur.
+                generatorInfo = generatorInfo.WithDocumentsAreFinalAndFrozen();
```

Comment:
pulled this to the location right before we create the finalstate.  that's much clearer to me.

Reply:
If nothing else, it really ensure we can't accidentally take a shortcut through the other method and skip that line.